### PR TITLE
document Grisu

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

File: base/grisu/grisu.jl [hunk 1]
```diff
@@ -26,6 +26,29 @@ include("grisu/bignum.jl")
 
 const BIGNUMS = [Bignums.Bignum(),Bignums.Bignum(),Bignums.Bignum(),Bignums.Bignum()]
 
+"""
+    (len, point, neg) = Grisu.grisu(v::AbstractFloat, mode, requested_digits,
+                buffer=DIGITSs[Threads.threadid()], bignums=BIGNUMS)
```
Comment (stevengj) [inline base/grisu/grisu.jl]: Needs to be updated for #27826?

File: base/grisu/grisu.jl [hunk 2]
```diff
@@ -26,6 +26,29 @@ include("grisu/bignum.jl")
...
+   digit). Note that this can be negative (for very small values), or greater than `len`
+   (for very large values).
+ - `neg`: the signbit of `v` (see [`signbit`](@ref)).
+
+`mode` can be one of:
+ - `Grisu.SHORTEST`: convert to the shortest decimal representation which can be "round-tripped" back to `v`.
+ - `Grisu.FIXED`: round to `requested_digits` digits.
+ - `Grisu.PRECISION`: round to `requested_digits` significant digits.
+
+`bignums` are used internally as part of the correction step.
```
Comment (stevengj) [inline base/grisu/grisu.jl]: `buffer` and `bignums`?

Comment (simonbyrne) [reply base/grisu/grisu.jl]: I mention above that `buffer` is used to output the digits

**Comment by simonbyrne** [comment]
I tweaked the documentation order to be inputs then outputs.

**Pull request merged**